From 14d29dec6285a1178f8e210ea0b9a8ccffa16c67 Mon Sep 17 00:00:00 2001
From: Song Bing <b06498@freescale.com>
Date: Tue, 15 Sep 2015 14:19:02 +0800
Subject: [PATCH 0450/5242] MLK-11542 media: mxc vout: reset video output
 count to zero when streamon

commit  3a7219d6857d5e1d1eb4cd21ea7dac3d0c734e0a from
https://source.codeaurora.org/external/imx/linux-imx.git

As streamoff/streamon will reset all buffer list, video output count should
be reset to zero.  Or timer will get wrong schedule time if input buffer time
stamp is 0.

Signed-off-by: Song Bing b06498@freescale.com
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/mxc/output/mxc_vout.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/platform/mxc/output/mxc_vout.c b/drivers/media/platform/mxc/output/mxc_vout.c
index 2871926..293b076 100644
--- a/drivers/media/platform/mxc/output/mxc_vout.c
+++ b/drivers/media/platform/mxc/output/mxc_vout.c
@@ -2075,6 +2075,8 @@ static int mxc_vidioc_streamon(struct file *file, void *fh,
 	hrtimer_init(&vout->timer, CLOCK_REALTIME, HRTIMER_MODE_ABS);
 	vout->timer.function = mxc_vout_timer_handler;
 	vout->timer_stop = true;
+	vout->frame_count = 0;
+	vout->vdi_frame_cnt = 0;
 
 	vout->start_ktime = hrtimer_cb_get_time(&vout->timer);
 
-- 
1.7.9.5

